tempest_parser/openstack_parse_tempest_html_page.py

```diff
@@ -0,0 +1,270 @@
+            with open(source, 'r', encoding='utf-8') as f:
+                content = f.read()
+            return content
+
+    def parse(self):
+        if not self.soup:
+            raise ValueError("No content loaded. Call load_content() first.")
+
+        self.failed_tests = []
+        self._parse_html_format()
+        return self.failed_tests
+
+    def _parse_html_format(self):
+        self._find_table_failures()
+
+    def _find_table_failures(self):
+        fail_links = self.soup.find_all('a', class_='popup_link')
+
+        for link in fail_links:
+            if link.text.strip().lower() != "fail":
+                continue
+       
+            row = link.find_parent('tr')
+            if not row:
+                continue
+ 
+            test_name = self._extract_full_test_name(row)
+ 
+            traceback = None
+            popup_id = link.get('href', '').split("'")[1] if "'" in link.get('href', '') else None
+
+            if popup_id:
+                popup_div = self.soup.find('div', id=popup_id)
+                if popup_div:
+                    pre_elem = popup_div.find('pre')
+                    if pre_elem:
+                        traceback = self._extract_traceback_from_text(pre_elem.text.strip())
+  
+            self.failed_tests.append((test_name, traceback or "No traceback found"))
+
+    def _extract_full_test_name(self, row):
+        test_case = ""
+  
+        test_case_div = row.find('div', class_='testcase')
+        if test_case_div:
+            test_case = test_case_div.text.strip()
+  
+        if not test_case:
+            return "Unknown test"
+
+        current = row
+        while current:
+            if current.get('class') and 'failClass' in current.get('class'):
+                testname_cell = current.find('td', class_='testname')
+                if testname_cell:
+                    suite_name = testname_cell.text.strip()
+                    return f"{suite_name}.{test_case}"
+    
+            current = current.find_previous_sibling('tr')
+
+        popup_id = None
+        popup_link = row.find('a', class_='popup_link')
+        if popup_link:
+            href = popup_link.get('href', '')
+            if 'showTestDetail' in href:
+                popup_id = href.split("'")[1] if "'" in href else None
+
+        if popup_id:
+            popup_div = self.soup.find('div', id=popup_id)
+            if popup_div:
+                pre_elem = popup_div.find('pre')
+                if pre_elem:
+                    content = pre_elem.text.strip()
+                    parts = content.split(':', 1)
+                    if len(parts) > 1 and '.' in parts[1]:
+                        full_name = parts[1].strip()
+                        if full_name.endswith(test_case):
+                            return full_name
+                        else:
+                            return full_name + '.' + test_case
+
+        return test_case
+
+    def _extract_traceback_from_text(self, text):
```

Comment:
I wrote a simpler parser here: https://github.com/RCAccelerator/chatbot/pull/126/files#diff-d1f0f72ecbf1b8b4722b31c67c986a63dfaf0614dcf4ee3c77022ba76b38b3deR96-R130

We can always compare results but it worked fine in my case, although we could improve my PR to return the test ID as well.

Reply:
@dsariel I think I'll try to take a look how we can use the @EmilienM's parser here. To ensure that we receive the same output of the parsing on the API side and on the side of populating of the qdrant as we've talked about.